client/src/components/CardDetail.js

```diff
@@ -0,0 +1,493 @@
+import React, { Fragment, useState } from "react";
+import {
+  FormGroup,
+  Dialog,
+  Button,
+  Typography,
+  Grid,
+  TextField,
+  InputBase,
+  DialogActions,
+  Select,
+  MenuItem,
+  Menu,
+  Checkbox,
+  Paper,
+  FormHelperText
+} from "@material-ui/core";
+import MuiDialogTitle from "@material-ui/core/DialogTitle";
+import MuiDialogContent from "@material-ui/core/DialogContent";
+import IconButton from "@material-ui/core/IconButton";
+import CloseIcon from "@material-ui/icons/Close";
+import Box from "@material-ui/core/Box";
+import { makeStyles, withStyles } from "@material-ui/core/styles";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import DateFnsUtils from "@date-io/date-fns";
+import { DateTimePicker, MuiPickersUtilsProvider } from "@material-ui/pickers";
+import { authFetch, authJSONFetch } from '../helpers/authFetch'
+import {useDashboard} from '../context/dashboard/dashboard.provider'
+
+const DialogTitle = (props) => {
+  const classes = makeStyles((theme) => ({
+    root: {
+      margin: 0,
+      padding: theme.spacing(4),
+      height: theme.spacing(4),
+    },
+    title: {
+      fontSize: 22,
+      display: "inline-block",
+      fontWeight: "bold",
+    },
+    closeButton: {
+      position: "absolute",
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+    colorLine: {
+      height: "7px",
+      width: "50px",
+      display: "inline-block",
+      borderRadius: "5px",
+      padding: "auto",
+      marginLeft: 10,
+    },
+  }))();
+
+  const { children, onClose, colorCode, setColorCode, ...other } = props;
+
+  const [anchorEl, setAnchorEl] = React.useState(null);
```

Comment:
1. you can directly useState instead of React.useState
2. can you rename the key to better suit the functionality, like maybe displayColorCode?